auth/rest_http_client.hh

```diff
@@ -0,0 +1,197 @@
+                auto bytes_authorization = bytes_view(reinterpret_cast<const signed char *>(authorization.c_str()),
+                                                      authorization.size());
+                return base64_encode(bytes_authorization);
+            }
+
+        public:
+            connection(connected_socket &&fd, sstring request)
+                    : _fd(std::move(fd)), _read_buf(_fd.input()), _write_buf(_fd.output()), _request(request) {}
+
+            ~connection() {}
+
+            future <role_set> do_get_groups(sstring username, sstring password) {
+                std::string b64_authorization = get_authorization_header(username, password);
+                sstring request = format(_request.c_str(), b64_authorization);
+
+                return _write_buf.write(request)
+                        .then([this] {
+                            return _write_buf.flush();
+                        })
+                        .then([this] {
+                            _parser.init();
+                            return _read_buf.consume(_parser);
+                        })
+                        .then([this] {
+                            if (_parser.eof()) {
+                                std::throw_with_nested(
+                                        exceptions::authentication_exception("Bad response: empty response"));
+                            }
+                            auto _rsp = _parser.get_parsed_response();
+
+                            switch (_rsp->_status) {
+                                case 200 :
+                                    return get_groups_from_body(std::move(_rsp));
+                                    break;
+                                case 401 :
+                                    std::throw_with_nested(exceptions::authentication_exception("Bad password"));
+                                    break;
+                                case 404 :
+                                    std::throw_with_nested(
+                                            exceptions::authentication_exception("Unknown username"));
+                                    break;
+                                default :
+                                    break;
+                            }
+                            std::throw_with_nested(exceptions::authentication_exception(
+                                    format("Issue to authenticate with http status code {}", _rsp->_status)));
+                        });
+
+            }
+        };
+
+        future<> init() {
+            _request = format("GET /api/v1/auth/user/groups HTTP/1.1\r\n"
+                              "Host: {}:{}\r\n"
+                              "Authorization: Basic {{}}\r\n"
+                              "Accept: application/json\r\n\r\n",
+                              _server, _port);
+
+            // Load system CA trust
+            // TODO see tls.credentials_builder in Seastar
```

Comment:
TODO reminder, do you still want to use builder ? current code seems ok to me.

Reply:
Idea behind that is to initialise a reloadable creds: tls::credentials_builder::build_reloadable_server_credentials (or at least look at it so) so we don't have to restart nodes if the ca is changed
But still not needed for V0

Reply:
ok I see, maybe just put this information in the comment